lib/bounces.js

```diff
@@ -108,6 +108,24 @@ module.exports = function (log, error) {
           }
         }
 
+        // Log flow metrics if `flowId` and `flowBeginTime` specified in headers
+        const flowId = getHeaderValue('X-Flow-Id', message)
+        const flowBeginTime = getHeaderValue('X-Flow-Begin-Time', message)
+        if (flowId && flowBeginTime) {
+          const eventName = 'email.' + templateName + '.bounced'
+          const currentTime = Date.now()
+
+          // Matches structure emitted by https://github.com/mozilla/fxa-auth-server/blob/master/lib/metrics/context.js#L122
+          const flowEventInfo = {
+            event: eventName,
+            time: currentTime,
+            flow_id: flowId,
+            flow_time: (currentTime <= flowBeginTime) ? 0 : (currentTime - flowBeginTime)
+          }
+
+          log.info('flowEvent', flowEventInfo)
```

Comment:
I agree with @philbooth, I think we probably want to instead call `log.flowEvent`, so that we have this logic in only 1 place. If it's cumbersome to create a fake request, that probably means we should update `log.flowEvent` to accept an alternative or easier structure.